src/main/java/org/takes/rq/RqRequestLine.java

```diff
@@ -0,0 +1,214 @@
+             * HTTPVERSION token.
+             */
+            HTTPVERSION(3);
+
+            /**
+             * Value.
+             */
+            private final int value;
+
+            /**
+             * Ctor.
+             * @param val Value
+             */
+            private Token(final int val) {
+                this.value = val;
+            }
+        }
+
+        /**
+         * Ctor.
+         * @param req Original request
+         */
+        public Base(final Request req) {
+            super(req);
+        }
+
+        @Override
+        public String requestLineHeader() throws IOException {
+            final String requestLine = this.getRequestLineHeader();
+            this.validateRequestLine(requestLine);
+            return requestLine;
+        }
+
+        @Override
+        public String method() throws IOException {
+            return this.requestLineHeaderToken(Token.METHOD);
+        }
+
+        @Override
+        public String requestUri() throws IOException {
+            return this.requestLineHeaderToken(Token.URI);
+        }
+
+        @Override
+        public String httpVersion() throws IOException {
+            return this.requestLineHeaderToken(Token.HTTPVERSION);
+        }
+
+        /**
+         * Get Request-Line header token.
+         * @param token Token
+         * @return HTTP Request-Line header token
+         * @throws IOException If fails
+         */
+        private String requestLineHeaderToken(final Token token)
+            throws IOException {
+            final String requestLine = this.getRequestLineHeader();
+            final Matcher matcher = this.validateRequestLine(requestLine);
+            final String result = matcher.group(token.value);
+            if (result == null) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        "There is no token %s in Request-Line header: %s",
+                        token.toString(),
+                        requestLine
+                    )
+                );
+            }
+            return result.trim();
+        }
+
+        /**
+         * Get Request-Line header.
+         *
+         * @return Valid Request-Line header
+         * @throws IOException If fails
+         */
+        private String getRequestLineHeader() throws IOException {
+            if (!this.head().iterator().hasNext()) {
+                throw new HttpException(
+                    HttpURLConnection.HTTP_BAD_REQUEST,
+                    "HTTP Request should have Request-Line"
+                );
+            }
+            return this.head().iterator().next();
+        }
+
+        /**
+         * Validate Request-Line according to PATTERN.
+         *
+         * @param requestline Request-Line header
+         * @return Matcher that can be used to extract tokens
+         * @throws HttpException If fails
+         */
+        private Matcher validateRequestLine(final String requestline)
```

Comment:
@xupyprmv it's a bad name for the method. it should be called `matcher`, not `validate`